### ISSUE TITLE
[Data Hookup]: Global validation for primary affiliation

### ISSUE THREAD

**Issue by kcullion** [icgc-argo/dac-ui#54]
## Goal
To hookup the primary affiliation validations that span sections A, B, and C.

## Expected Outcomes

- [x] **Applicant, Institutional Rep, Collaborators all have to belong to the same primary affiliation**. i.e. the Primary Affiliation filled out in these sections must all be exactly the same (section A is the source of truth)

1. If Section A primary affiliation is filled out, for section B and any collaborator added in section C - those primary affiliation values must match exactly. If they don't then show the error (listing the Primary Affiliation of section A within the error text)
![image](https://user-images.githubusercontent.com/18424138/122471216-fbb13d80-cf8c-11eb-9823-c7ebb87d44db.png)

2. If Section A primary affiliation is not filled out, and the applicant fills out the primary affiliation in section B, don't show any error. Then if they go back to section A and fill out a different primary affiliation,  show the error in the section B primary affiliation field (listing the Primary Affiliation of section A within the error text).
![image](https://user-images.githubusercontent.com/18424138/122471706-93af2700-cf8d-11eb-8a6c-d20dd3c37f7a.png)

3. If section A primary affiliation is not filled out, and the applicant adds a collaborator in section C, they can add that collaborator for any primary affiliation they fill out. If the applicant then fills out a different primary affiliation in section A than what was added for that collaborator, show the error in **section C** that primary affiliations much match. 
**So section table would look like this (rows of table with errors are shaded red) **: Zeplin:  https://zpl.io/amLBY43
![image](https://user-images.githubusercontent.com/18424138/120340553-22e8e900-c2c4-11eb-90e4-635cffc0aa90.png)

**When user clicks the edit pencil for that collaborator, the edit modal would look like:**

- [x] Show the error on Primary affiliation
- [x] disable the save button until they enter the correct afffiliation

![image](https://user-images.githubusercontent.com/18424138/122470336-f0114700-cf8b-11eb-98b0-e2fbe1a6e491.png)

**Comment by rosibaj** [comment]
@kcullion
for section A - i think it makes sense that Section A is the source of truth, since that is the applicant section.  the primary affiliation should be where the applicant is from (not a collaborator)!
for section B - if you select `Same as applicant` should that section just be disabled and the user just fills out section A?

**Comment by kcullion** [comment]
> @kcullion
> for section A - i think it makes sense that Section A is the source of truth, since that is the applicant section. the primary affiliation should be where the applicant is from (not a collaborator)!
> for section B - if you select `Same as applicant` should that section just be disabled and the user just fills out section A?

Thanks @rosibaj can you review my changes?

**Comment by kcullion** [comment]
Thanks @caravinci ... it's working well!

Just found this thing: 
- [ ] I filled out the Primary Affiliation (PA) in section B as oi`c`r. Then I filled out `google` in section A. I see there is an error in section B but I have to refresh to see the actual error message. I should see that message when going back to section B or else I may not refresh and I may not know what to fix.
eg. it looks like this section is correctly filled out but it does have an error
![image](https://user-images.githubusercontent.com/18424138/127373859-cd34ac5e-ba71-455c-add0-bff3a1d7f0bf.png)
